### PR TITLE
read entrainment parameters from param_set

### PR DESCRIPTION
**Pull request by yairchn**

### CODE REVIEW

File: integration_tests/utils/parameter_set.jl [hunk 1]
```diff
@@ -29,6 +36,14 @@ function create_parameter_set(namelist)
         α_d = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "pressure_normalmode_drag_coeff"),
         H_up_min = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "min_updraft_top"),
         ω_pr = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "Prandtl_number_scale"),
+        c_δ = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "detrainment_factor"),
+        β = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "sorting_power"),
+        χ = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "updraft_mixing_frac"),
+        c_t = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "turbulent_entrainment_factor"),
+        c_λ = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "entrainment_smin_tke_coeff"),
+        w_min = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "min_upd_velocity"),
```
Comment (haakon-e) [inline integration_tests/utils/parameter_set.jl:44]: should we specify that this is an updraft velocity? e.g. `w_up_min`?

File: integration_tests/utils/parameter_set.jl [hunk 2]
```diff
@@ -15,6 +15,13 @@ CLIMAParameters.Atmos.EDMF.α_a(ps::EarthParameterSet) = ps.nt.α_a
 CLIMAParameters.Atmos.EDMF.α_d(ps::EarthParameterSet) = ps.nt.α_d
 CLIMAParameters.Atmos.EDMF.H_up_min(ps::EarthParameterSet) = ps.nt.H_up_min
 CLIMAParameters.Atmos.EDMF.ω_pr(ps::EarthParameterSet) = ps.nt.ω_pr
+CLIMAParameters.Atmos.EDMF.c_δ(ps::EarthParameterSet) = ps.nt.c_δ
+CLIMAParameters.Atmos.EDMF.β(ps::EarthParameterSet) = ps.nt.β
+CLIMAParameters.Atmos.EDMF.χ(ps::EarthParameterSet) = ps.nt.χ
+CLIMAParameters.Atmos.EDMF.c_t(ps::EarthParameterSet) = ps.nt.c_t
+CLIMAParameters.Atmos.EDMF.c_λ(ps::EarthParameterSet) = ps.nt.c_λ
+CLIMAParameters.Atmos.EDMF.w_min(ps::EarthParameterSet) = ps.nt.w_min
+CLIMAParameters.Atmos.EDMF.μ_0(ps::EarthParameterSet) = ps.nt.μ_0
```
Comment (haakon-e) [inline integration_tests/utils/parameter_set.jl:24]: could you sort and document what types of closures these belong to?

File: integration_tests/utils/parameter_set.jl [hunk 3]
```diff
@@ -29,6 +36,14 @@ function create_parameter_set(namelist)
...
         H_up_min = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "min_updraft_top"),
         ω_pr = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "Prandtl_number_scale"),
+        c_δ = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "detrainment_factor"),
+        β = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "sorting_power"),
+        χ = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "updraft_mixing_frac"),
+        c_t = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "turbulent_entrainment_factor"),
+        c_λ = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "entrainment_smin_tke_coeff"),
+        w_min = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "min_upd_velocity"),
+        μ_0 = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "entrainment_scale"),
+        μ = TC.parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "entrainment_sigma"),
```
Comment (haakon-e) [inline integration_tests/utils/parameter_set.jl:46]: same here. add some documentation for which closures the parameters relate to.

File: src/closures/entr_detr.jl [hunk 4]
```diff
@@ -3,35 +3,25 @@
...
 
+    l = zeros(2)
     c_ε = CPEDMF.c_ε(param_set)
+    c_δ = CPEDMF.c_δ(param_set)
+    c_t = CPEDMF.c_t(param_set)
+    c_λ = CPEDMF.c_λ(param_set)
+    w_min = CPEDMF.w_min(param_set)
     c_div = ICP.entrainment_massflux_div_factor(param_set)
 
+    # should be: c_δ = sign(condensate(ts_en) + condensate(ts_up[i])) * entr.c_δ
```
Comment (haakon-e) [inline src/closures/entr_detr.jl:24]: is this a TODO comment for a future PR?

File: src/closures/nondimensional_exchange_functions.jl [hunk 5]
```diff
@@ -13,22 +9,22 @@
 
 Returns the nondimensional entrainment and detrainment
 functions following Cohen et al. (JAMES, 2020), given:
- - `param_set` a parameter set
- - `c_δ`,  detrainment factor
- - `c_μ`,  logistic function factor
- - `c_μ0`, logistic function timescale (sec)
- - `β`,    sorting power
- - `χ_up`, updraft mixing fraction
+ - `μ`,  logistic function factor
```
Comment (haakon-e) [inline src/closures/nondimensional_exchange_functions.jl:12]: you deleted `param_set` docs, but added `μ` docs. Should keep `param_set` and remove `μ`, no?

**Review by haakon-e — COMMENTED** [review]
Looks like a good start

File: integration_tests/utils/parameter_set.jl (hunk 3 above)
Comment (yairchn) [reply integration_tests/utils/parameter_set.jl:46]: there is one microphysics parameter that I need to ask Anna but otherwise I added them

**Review by costachris — APPROVED** [review]

File: src/closures/nondimensional_exchange_functions.jl (hunk 5 above)
Comment (yairchn) [reply src/closures/nondimensional_exchange_functions.jl]: that was a rebasing error - fixed. good catch!

File: integration_tests/utils/parameter_set.jl [hunk 6]
```diff
@@ -9,12 +9,19 @@ end
...
-CLIMAParameters.Atmos.EDMF.c_ε(ps::EarthParameterSet) = ps.nt.c_ε
-CLIMAParameters.Atmos.EDMF.α_b(ps::EarthParameterSet) = ps.nt.α_b
-CLIMAParameters.Atmos.EDMF.α_a(ps::EarthParameterSet) = ps.nt.α_a
-CLIMAParameters.Atmos.EDMF.α_d(ps::EarthParameterSet) = ps.nt.α_d
-CLIMAParameters.Atmos.EDMF.H_up_min(ps::EarthParameterSet) = ps.nt.H_up_min
-CLIMAParameters.Atmos.EDMF.ω_pr(ps::EarthParameterSet) = ps.nt.ω_pr
+CLIMAParameters.Atmos.EDMF.c_ε(ps::EarthParameterSet) = ps.nt.c_ε # factor multiplyer for dry term in entrainment/detrainment
+CLIMAParameters.Atmos.EDMF.α_b(ps::EarthParameterSet) = ps.nt.α_b # factor multiplyer for pressure buoyancy terms (effective buoyancy is (1-α_b))
+CLIMAParameters.Atmos.EDMF.α_a(ps::EarthParameterSet) = ps.nt.α_a # factor multiplyer for pressure advection
+CLIMAParameters.Atmos.EDMF.α_d(ps::EarthParameterSet) = ps.nt.α_d # factor multiplyer for pressure drag
```
Comment (haakon-e) [inline integration_tests/utils/parameter_set.jl:15]: Looks good. I was thinking you could also simply add something like
```suggestion
# entrainment/detrainment parameters
CLIMAParameters.Atmos.EDMF.c_ε(ps::EarthParameterSet) = ps.nt.c_ε # factor multiplyer for dry term in entrainment/detrainment
CLIMAParameters.Atmos.EDMF.α_b(ps::EarthParameterSet) = ps.nt.α_b # factor multiplyer for pressure buoyancy terms (effective buoyancy is (1-α_b))
CLIMAParameters.Atmos.EDMF.α_a(ps::EarthParameterSet) = ps.nt.α_a # factor multiplyer for pressure advection
CLIMAParameters.Atmos.EDMF.α_d(ps::EarthParameterSet) = ps.nt.α_d # factor multiplyer for pressure drag
```

File: src/closures/entr_detr.jl (hunk 4 above)
Comment (yairchn) [reply src/closures/entr_detr.jl:24]: yes for when we have thermodynamics.jl implemented and we can all condensate(ts) to see if there is any

**Review by haakon-e — APPROVED** [review]

**Pull request merged**